pkg/apis/sources/v1alpha1/mongodbsource_validation.go

```diff
@@ -33,16 +35,15 @@ func (m *MongoDbSource) Validate(ctx context.Context) *apis.FieldError {
 	return errs
 }
 
-// Validate validates MongoDbSourceSpecs
+// Validate validates MongoDbSourceSpecs.
 func (ms *MongoDbSourceSpec) Validate(ctx context.Context) *apis.FieldError {
 	var errs *apis.FieldError
 
-	// Validate sink
-	if ms.Sink == nil {
-		fe := apis.ErrMissingField("sink")
-		errs = errs.Also(fe)
-	} else if fe := ms.Sink.Validate(ctx); fe != nil {
-		errs = errs.Also(fe.ViaField("sink"))
+	// Validate sink.
+	if equality.Semantic.DeepEqual(ms.Sink, duckv1.Destination{}) {
```

Comment:
no need for this empty one here. 

Follow this validation template: https://github.com/knative-sandbox/sample-source/blob/master/pkg/apis/samples/v1alpha1/samplesource_validation.go

And we should validate the secret as well. 
You can copy this helper method to do so.
https://github.com/google/knative-gcp/blob/5bdd3d75c6f93107681a0ad4319dd8278708bdd7/pkg/apis/intevents/v1beta1/pullsubscription_validation.go#L106